subprojects/docs/src/docs/userguide/idea_plugin.adoc

```diff
@@ -152,69 +154,76 @@ The plugin provides objects modeling the sections of the metadata files that are
 +
- * `beforeMerged { api:org.gradle.plugins.ide.idea.model.Project[] arg \-> ... }`
- * `whenMerged { api:org.gradle.plugins.ide.idea.model.Project[] arg \-> ... }`
- * `withXml { api:org.gradle.api.XmlProvider[] arg \-> ... }`
+ * `beforeMerged { link:{javadocPath}/org/gradle/plugins/ide/idea/model/Project.html[Project] arg \-> ... }`
+ * `whenMerged { link:{javadocPath}/org/gradle/plugins/ide/idea/model/Project.html[Project] arg \-> ... }`
+ * `withXml { link:{javadocPath}/org/gradle/api/XmlProvider.html[XmlProvider] arg \-> ... }`
 
-api:org.gradle.plugins.ide.idea.model.IdeaModule[]::
- * `beforeMerged { api:org.gradle.plugins.ide.idea.model.Module[] arg \-> ... }`
- * `whenMerged { api:org.gradle.plugins.ide.idea.model.Module[] arg \-> ... }`
- * `withXml { api:org.gradle.api.XmlProvider[] arg \-> ... }`
+link:{javadocPath}/org/gradle/plugins/ide/idea/model/IdeaModule.html[IdeaModule]::
+ * `beforeMerged { link:{javadocPath}/org/gradle/plugins/ide/idea/model/Module.html[Module] arg \-> ... }`
+ * `whenMerged { link:{javadocPath}/org/gradle/plugins/ide/idea/model/Module.html[Module] arg \-> ... }`
+ * `withXml { link:{javadocPath}/org/gradle/api/XmlProvider.html[XmlProvider] arg \-> ... }`
 
-api:org.gradle.plugins.ide.idea.model.IdeaWorkspace[]::
- * `beforeMerged { api:org.gradle.plugins.ide.idea.model.Workspace[] arg \-> ... }`
- * `whenMerged { api:org.gradle.plugins.ide.idea.model.Workspace[] arg \-> ... }`
- * `withXml { api:org.gradle.api.XmlProvider[] arg \-> ... }`
+link:{javadocPath}/org/gradle/plugins/ide/idea/model/IdeaWorkspace.html[IdeaWorkspace]::
+ * `beforeMerged { link:{javadocPath}/org/gradle/plugins/ide/idea/model/Workspace.html[Workspace] arg \-> ... }`
+ * `whenMerged { link:{javadocPath}/org/gradle/plugins/ide/idea/model/Workspace.html[Workspace] arg \-> ... }`
+ * `withXml { link:{javadocPath}/org/gradle/api/XmlProvider.html[XmlProvider] arg \-> ... }`
 
 
 [[sec:partial-rewrite]]
-===== Partial rewrite of existing content
+==== Partial rewrite of existing content
+
+A "complete rewrite" causes all existing content to be discarded, thereby losing any changes made directly in the IDE. The `beforeMerged` hook makes it possible to overwrite just certain parts of the existing content. The following example removes all existing dependencies from the `Module` domain object:
 
-A <<sec:complete-rewrite,complete rewrite>> causes all existing content to be discarded, thereby losing any changes made directly in the IDE. The `beforeMerged` hook makes it possible to overwrite just certain parts of the existing content. The following example removes all existing dependencies from the `Module` domain object:
+=== Example: Partial Rewrite for Module
 
-++++
-<sample id="partialRewrites" dir="idea" title="Partial Rewrite for Module">
-    <sourcefile file="build.gradle" snippet="module-before-merged"/>
-</sample>
-++++
+[source.multi-language-sample,groovy]
```

Comment:
❌ The sample has a weird highlighting: the `module` and `true` words are presented with blue color.

Reply:
module is now a java keyword that's inherited by the syntax highlighter I'm guessing. I'd rather address syntax highlighting nuances separately if you don't mind.

Reply:
Sure, it's a minor detail. I just wanted to mention in case a custom formatting was missing.